nexus/src/db/datastore.rs

```diff
@@ -1011,24 +1182,66 @@ impl DataStore {
         &self,
         opctx: &OpContext,
         disk_authz: authz::ProjectChild,
+    ) -> DeleteResult {
```

Comment:
As far as I can tell, this function has no callers, so I think you can remove it.  (It would be weird for there to be any callers because you need a saga to delete a disk, and the saga calls the internal version.)